packages/app/src/cli/prompts/deploy-release.ts

```diff
@@ -0,0 +1,184 @@
+import metadata from '../metadata.js'
+import {
+  ConfigExtensionIdentifiersBreakdown,
+  ExtensionIdentifiersBreakdown,
+} from '../services/context/breakdown-extensions.js'
+import {useVersionedAppConfig} from '@shopify/cli-kit/node/context/local'
+import {InfoTableSection, renderConfirmationPrompt, renderDangerousConfirmationPrompt} from '@shopify/cli-kit/node/ui'
+
+export interface DeployOrReleaseConfirmationPromptOptions {
+  extensionIdentifiersBreakdown: ExtensionIdentifiersBreakdown
+  configExtensionIdentifiersBreakdown?: ConfigExtensionIdentifiersBreakdown
+  appTitle?: string
+  release: boolean
+  force: boolean
+}
+
+export interface DeployConfirmationPromptOptions {
+  appTitle?: string
+  extensionsContentPrompt: {
+    extensionsInfoTable?: InfoTableSection
+    deletedInfoTable?: InfoTableSection
+  }
+  configContentPrompt: {
+    configInfoTable: InfoTableSection
+    deletedInfoTable?: InfoTableSection
+  }
+  release: boolean
+}
+
+export async function deployOrReleaseConfirmationPrompt({
+  force,
+  extensionIdentifiersBreakdown,
+  configExtensionIdentifiersBreakdown,
+  appTitle,
+  release,
+}: DeployOrReleaseConfirmationPromptOptions) {
+  if (force) return true
+  const extensionsContentPrompt = await buildExtensionsContentPrompt(extensionIdentifiersBreakdown)
+  const configContentPrompt = await buildConfigContentPrompt(release, configExtensionIdentifiersBreakdown)
```

Comment:
This does not have to be an async function

Reply:
yep, I will address this refactor in the final PR